### PR TITLE
Add unit tests for com.mars.core.util.FileUtil

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that `com.mars.core.util.FileUtil` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    @Test
+    public void testReadYml() throws Exception {
+        PowerMockito.mockStatic(Yaml.class);
+        PowerMockito.mockStatic(JSON.class);
+
+        Method toJSONStringMethod = DTUMemberMatcher.method(
+                JSON.class, "toJSONString", Object.class);
+        PowerMockito.doReturn("foo")
+                .when(JSON.class, toJSONStringMethod)
+                .withArguments(or(isA(Object.class), isNull(Object.class)));
```
Comment (JohnLBergqvist) [inline mars-core/src/test/java/com/mars/core/util/FileUtilTests.java:66]: in practice you don't need the `or` annotation: `.withArguments(isA(Object.class))`

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        Assert.assertEquals("foo", FileUtil.readYml("a/b/c"));
+    }
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
```
Comment (JohnLBergqvist) [inline mars-core/src/test/java/com/mars/core/util/FileUtilTests.java]: I'd put this at the top of the class, before any tests

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        FileUtil.readYml(null);
+    }
+
+    @PrepareForTest({FileInputStream.class, FileUtil.class, File.class})
+    @Test
+    public void testGetFileToByte1() throws Exception {
+        File file = PowerMockito.mock(File.class);
+        FileInputStream fileInputStream =
+                PowerMockito.mock(FileInputStream.class);
+        PowerMockito.when(fileInputStream.read(or(isA(byte[].class),
```
Comment (JohnLBergqvist) [inline mars-core/src/test/java/com/mars/core/util/FileUtilTests.java:83]: Again, you don't need the `or` here.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java (hunk 1 above)
Comment (Braavos6) [reply mars-core/src/test/java/com/mars/core/util/FileUtilTests.java:66]: I tried to take it out and then it give me this error `java.lang.Exception: No tests found matching Method testReadYml(com.mars.core.util.FileUtilTests)`

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java (hunk 2 above)
Comment (Braavos6) [reply mars-core/src/test/java/com/mars/core/util/FileUtilTests.java]: Done

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java (hunk 1 above)
Comment (JohnLBergqvist) [reply mars-core/src/test/java/com/mars/core/util/FileUtilTests.java:66]: Weird.

**Review by JohnLBergqvist — APPROVED** [review]

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.junit.rules.ExpectedException;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import java.awt.image.BufferedImage;
+import java.io.*;
```
Comment (paulkennethkent) [inline mars-core/src/test/java/com/mars/core/util/FileUtilTests.java]: is the wildcard necessary here?

**Review by paulkennethkent — APPROVED** [review]

File: mars-core/src/test/java/com/mars/core/util/FileUtilTests.java (hunk 4 above)
Comment (Braavos6) [reply mars-core/src/test/java/com/mars/core/util/FileUtilTests.java]: removed.. i didn't see that.. it must have been an auto import

**Pull request closed without merging**